src/ring_theory/polynomial/hermite.lean

```diff
@@ -50,4 +51,71 @@ begin
   simp only [map_one, mul_one, derivative_one, sub_zero]
 end
 
+section coeff
+
+lemma coeff_hermite_succ_zero (n : ℕ) :
+  coeff (hermite (n + 1)) 0 = -(coeff (hermite n) 1) := by simp [coeff_derivative]
+
+lemma coeff_hermite_succ_succ (n k : ℕ) :
+  coeff (hermite (n + 1)) (k + 1) = coeff (hermite n) k - (k + 2) * (coeff (hermite n) (k + 2)) :=
+begin
+  rw [hermite_succ, coeff_sub, coeff_X_mul, coeff_derivative, mul_comm],
+  norm_cast
+end
+
+lemma hermite_coeff_upper (n k : ℕ) : coeff (hermite n) (n + k + 1) = 0 :=
+begin
+  induction n with n ih generalizing k,
+  { apply coeff_C },
+  {  rw [coeff_hermite_succ_succ,
+      (by linarith : n + 1 + k = n + k + 1),
+      (by linarith : n + k + 1 + 2 = n + (k + 2) + 1),
+      ih k, ih (k + 2), mul_zero, sub_zero] }
+end
+
+lemma hermite_coeff_upper' (n k : ℕ) (hnk : n < k) : coeff (hermite n) k = 0 :=
+begin
+  obtain ⟨k, rfl⟩ := nat.exists_eq_add_of_lt hnk,
+  apply hermite_coeff_upper
+end
+
+lemma coeff_hermite_self (n : ℕ) : coeff (hermite n) n = 1 :=
+begin
+  induction n with n ih,
+  { apply coeff_C },
+  { rw [coeff_hermite_succ_succ, ih, hermite_coeff_upper, mul_zero, sub_zero] }
+end
+
+@[simp] lemma degree_hermite (n : ℕ) : (hermite n).degree = n :=
+begin
+  rw degree_eq_of_le_of_coeff_ne_zero,
+  simp_rw [degree_le_iff_coeff_zero, with_bot.coe_lt_coe],
+  exact hermite_coeff_upper' n,
+  simp [coeff_hermite_self n]
```

Comment:
Either use curly parentheses or inline `hermite_coeff_upper' n`. I wonder whether `simp [degree_le_iff_coeff_zero, with_bot.coe_lt_coe, hermite_coeff_upper' n, coeff_hermite_self n]` proves everything in one line.